aten/src/ATen/native/layer_norm.cpp

```diff
@@ -0,0 +1,647 @@
+#include <ATen/NativeFunctions.h>
```

Comment:
A lot of this stuff can be rewritten using TensorIterator or existing abstractions such as "mean_std". We need to sit down and write up more docs on those.

Reply:
Here is the `mean_std` PR with implementation for reference https://github.com/pytorch/pytorch/pull/18731